src/single_translator_VAE/tl/single_cell_tools.py

```diff
@@ -0,0 +1,393 @@
+import os
+import sys
+from collections import Counter
+
+import numpy as np
+import pandas as pd
+import ray as ray
+import scanpy as sc
+import torch
+import torch.nn
+from scipy.stats import poisson
+
+import single_translator_VAE as sv
+
+sys.path.insert(1, "../../")
+sys.path.insert(1, "../")
+sys.path.insert(1, "../../../../../")
+
+
+def split_ID(row):
+    """Takes in one row of DataFrame, returns the right cell name in metadata of MBC"""
+    return row.split("-")[4]
+
+
+def split_ID_2(row):
+    """Takes in one row of DataFrame, returns the right cell name in metadata of MBC"""
+    return row.split("-")[-1]
+
+
+def merge_strings(row):
+    """Takes in one row of DataFrame, returns the merged strings"""
+    return row + "-1"
+
+
+def prep_files_for_deconvolution(bulks, reference, path, file_name, deconvolution_method, genes, cell_types):
+    """
+    Prepare and save the data for deconvolution methods.
+
+    Parameters
+    ----------
+    bulks (Pandas dataframe): The bulk data.
+    reference (AnnData): The reference data.
+    path (str): The directory where the data should be saved.
+    file_name (str): The base name of the file.
+    deconvolution_method (str): The deconvolution method ('bayesprism', 'nnls', 'cibersortx').
+
+    Returns
+    -------
+    dict: Paths of the created data files.
+    """
+    genes = reference.var.index
+    cell_types = reference.obs.cell_types.unique()
+    reference = reference.X.toarray()
+
+    method_path = os.path.join(path, deconvolution_method, "input_data")
+    if not os.path.exists(method_path):
+        os.makedirs(method_path)
+
+    file_paths = {}
+
+    if deconvolution_method == "cibersortx":
```

Comment:
in python 3.10 theres a new way to do switch statements. You don't have to change to this syntax, just an FYI.

https://www.geeksforgeeks.org/python-match-case-statement/

Reply:
nice, i changed it